executor/ddl.go

```diff
@@ -77,12 +112,38 @@ func (e *DDLExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	e.done = true
 
 	// For each DDL, we should commit the previous transaction and create a new transaction.
-	// An exception is create local temporary table.
-	if s, ok := e.stmt.(*ast.CreateTableStmt); ok {
+	// Following cases are exceptions
+	switch s := e.stmt.(type) {
+	case *ast.CreateTableStmt:
 		if s.TemporaryKeyword == ast.TemporaryLocal {
 			return e.createSessionTemporaryTable(s)
 		}
+	case *ast.DropTableStmt:
+		if s.IsView {
+			break
+		}
+		sessVars := e.ctx.GetSessionVars()
+		sessVarsTempTable := sessVars.LocalTemporaryTables
+		if sessVarsTempTable == nil {
+			break
+		}
+		localTempTables := sessVarsTempTable.(*infoschema.LocalTemporaryTables)
+		var existsNum int
+		for tbIdx := range s.Tables {
```

Comment:
Some advices to make the code more simple:

1. Prepare a variable `var localTemporaryTable []*model.TableInfo` before this switch clause.
2. Iter s.Tables, if current table is a local temporary table, remove it from s.Tables, and push it to localTemporaryTable
3. After that `len(s.Tables)==0` , you can drop all temporary tables directly. If not normal table should be dropped in next switch.
4. You do not need to modify any code in `dropTableObject` because all tables passed to it can be regarded as normal table.
5.  You just need to invoke `dropLocalTemporaryTables` after `executeDropTable`